python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1836 @@
+            Azimuth angle for the dome (in deg).
+        """
+        await salobj.set_summary_state(self.rem.atdometrajectory, salobj.State.DISABLED)
+
+        self.rem.atdome.evt_azimuthInPosition.flush()
+
+        target_az = Angle(az, unit=u.deg).deg
+        await self.rem.atdome.cmd_moveAzimuth.set_start(
+            azimuth=target_az, timeout=self.long_long_timeout
+        )
+
+        self.check.atmcs = False
+        self.check.atdometrajectory = False
+        self.check.atdome = True
+
+        task_list = [
+            asyncio.create_task(
+                self.wait_for_inposition(timeout=self.long_long_timeout)
+            ),
+            asyncio.create_task(self.monitor_position()),
+            asyncio.create_task(
+                self.check_component_state("atdometrajectory", salobj.State.DISABLED)
+            ),
+        ]
+
+        try:
+            await self.process_as_completed(task_list)
+        except Exception as e:
+            self.check.atmcs = True
+            self.check.atdometrajectory = True
+            raise e
+
+    async def prepare_for_flatfield(self):
+        """A high level method to position the telescope and dome for flat
+        field operations.
+
+        The method will,
+
+            1 - disable ATDomeTrajectory
+            2 - send telescope to flat field position
+            3 - send dome to flat field position
+            4 - re-enable ATDomeTrajectory
+        """
+        await salobj.set_summary_state(self.rem.atdometrajectory, salobj.State.DISABLED)
+
+        await self.open_m1_cover()
+
+        atdometrajectory_check = self.check.atdometrajectory
+        self.check.atdometrajectory = False
+
+        try:
+            await self.point_azel(
+                target_name="FlatField position",
+                az=self.tel_flat_az,
+                el=self.tel_flat_el,
+                wait_dome=False,
+            )
+
+            try:
+                await self.stop_tracking()
+            except asyncio.TimeoutError:
+                self.log.debug("Timeout in stopping tracking. Continuing.")
+
+            await self.slew_dome_to(self.dome_flat_az)
+        except Exception as e:
+            self.check.atdometrajectory = atdometrajectory_check
+            raise e
+        finally:
+            await salobj.set_summary_state(
+                self.rem.atdometrajectory, salobj.State.ENABLED
+            )
+
+    async def stop_tracking(self):
+        """Task to stop telescope tracking."""
+
+        self.log.debug("Stop tracking.")
+
+        self.rem.atmcs.evt_atMountState.flush()
+        self.rem.atmcs.evt_allAxesInPosition.flush()
+
+        await self.rem.atptg.cmd_stopTracking.start(timeout=self.fast_timeout)
+
+        try:
+            at_mount_state = await self.rem.atmcs.evt_atMountState.aget(
+                timeout=self.long_timeout
+            )
+        except asyncio.TimeoutError:
+            self.log.warning("Timeout waiting for atMountState event.")
```

Comment:
Please document the problem you are working around and provide a ticket to get rid of this workaround. (Normally if it timed out you'd fail!)